packages/zoe/src/contractFacet/contractFacet.js

```diff
@@ -297,6 +317,17 @@ export function buildRootObject() {
           seatData,
           getAmountMath,
         );
+
+        E(zoeInstanceAdmin)
+          .makeOfferlessSeat(initialAllocation, proposal)
+          .then(({ zoeSeatAdmin, notifier: zoeNotifier, userSeat }) => {
+            updateFromNotifier(updater, zoeNotifier);
+            zoeSeatAdminPromiseKit.resolve(zoeSeatAdmin);
+            userSeatPromiseKit.resolve(userSeat);
+
+            seatToZoeSeatAdmin.init(zcfSeat, zoeSeatAdmin);
```

Comment:
Here's where we have the same problem in a form that 
   * correctly passes our type checker,
   * is unsurprisingly undetected by our runtime tests
   * that I completely missed on first careful review of the code.

In order to register a presence, rather than a promise, this only registers eventually. However, a reallocate may happen before this registration, causing the lookup to happen, and fail, immediately, blocking a legitimate reallocate from succeeding.

I caught this only because my refactoring attempt refactored this problem into a more obvious bug. Once hit, I asked myself "Now wait a second, why doesn't the original have this problem?".

I don't see a straightforward way to fix this.

Reply:
Hmm, ok, that is a good point. Let me check my understanding of how our ordering works: I can guarantee that the `E(zoeInstanceAdmin).makeOfferlessSeat` call will be received by the Zoe Service vat before the E(zoeInstanceAdmin).replaceAllocations` call, right? But I can't guarantee that I will get a response from `E(zoeInstanceAdmin).makeOfferlessSeat` before the `E(zoeInstanceAdmin).replaceAllocations` call is made, right?

I think a way to fix this might be to go back to `offerHandles` (`seatHandles`) internally, which can be originated on either the Zoe or ZCF side, but which can be used for identification rather than having to rely on objects that need information (sometimes information from promises resolving) to be made.

Reply:
It seems that `makeEmptySeatKit()` shouldn't return control to the contract until it has a response from zoe on the `makeOfferlessSeat()` call.

Reply:
> It seems that `makeEmptySeatKit()` shouldn't return control to the contract until it has a response from zoe on the `makeOfferlessSeat()` call.

I think we can't do that because we need the call to be synchronous from the contract side into ZCF.

Reply:
@katelynsills I do not understand your suggested `seatHandle`s fix. Rather than spend more time on this today, let's all keep in mind that the atomicity problem that this PR fixes, #1391 , is only a problem in theory for us now (see the analysis there). So, for a potential merge of #1414 into master I'd rather live with the status quo that spend further time today trying to rescue this PR.

Reply:
Ok, I'm ok putting this off until the reviews are done.